rgw/v2/tests/s3_swift/test_multisite_syncpolicy_prefix_tag.py

```diff
@@ -0,0 +1,182 @@
+    basic_io_structure = BasicIOInfoStructure()
+    io_info_initialize.initialize(basic_io_structure.initial())
+    user_info = s3lib.create_users(config.user_count)
+
+    # Create Zonegroup policy
+    log.info("Creating Zone group policy at Enabled state")
+    group_id = "global_group"
+    reusable.group_operation(group_id, "create", "enabled")
+    reusable.flow_operation(group_id, "create")
+    reusable.pipe_operation(group_id, "create")
+
+    for user in user_info:
+        auth = Auth(user, ssh_con, ssl=config.ssl)
+        rgw_conn = auth.do_auth()
+
+        for bc in range(config.bucket_count):
+            bucket_name = utils.gen_bucket_name_from_userid(user["user_id"], rand_no=bc)
+            bucket = reusable.create_bucket(bucket_name, rgw_conn, user)
+            log.info(f"Bucket {bucket_name} created")
+            prefix = "foo"
+            obj_tag = '{"TagSet":[{"Key":"colour", "Value":"red"}]}'
+            # Create bucket sync policy
+            group_id1 = "group-" + bucket_name
+            reusable.group_operation(
+                group_id1,
+                "create",
+                "enabled",
+                bucket_name,
+            )
+            if config.test_ops["has_prefix"] and config.test_ops["has_tag"]:
+                detail = f" --prefix={prefix} --tags-add={obj_tag}"
+            elif (
+                config.test_ops["has_prefix"] is True
+                and config.test_ops["has_tag"] is False
+            ):
+                detail = f" --prefix={prefix}"
+            elif (
+                config.test_ops["has_prefix"] is False
+                and config.test_ops["has_tag"] is True
+            ):
+                detail = f" --tags-add={obj_tag}"
+            log.info("Creating bucket policy with details")
+            pipe_id = reusable.pipe_operation(
+                group_id1,
+                "create",
+                bucket_name=bucket_name,
+                policy_detail=detail,
+            )
+            time.sleep(60)
+
+            log.info(f"Creating objects on bucket: {bucket_name}")
+            log.info("s3 objects to create: %s" % config.objects_count)
+
+            for oc, size in list(config.mapped_sizes.items()):
+                config.obj_size = size
+                s3_object_name = utils.gen_s3_object_name(bucket_name, oc)
+                if config.test_ops["has_prefix"]:
+                    # adding prefix
+                    s3_object_name = prefix + s3_object_name
+                    log.info("s3 object name: %s" % s3_object_name)
+                    s3_object_path = os.path.join(TEST_DATA_PATH, s3_object_name)
+                    log.info("s3 object path: %s" % s3_object_path)
+                else:
+                    log.info("s3 object name: %s" % s3_object_name)
+                    s3_object_path = os.path.join(TEST_DATA_PATH, s3_object_name)
+                    log.info("s3 object path: %s" % s3_object_path)
```

Comment:
could you simplify these lines as discussed

Reply:
updated